wlroots/wlr_types/data_device_manager.py

```diff
@@ -13,3 +15,84 @@ def __init__(self, display: Display) -> None:
             The display to handle the clipboard for
         """
         self._ptr = lib.wlr_data_device_manager_create(display._ptr)
+
+
+class Drag(Ptr):
+    def __init__(self, ptr) -> None:
+        self._ptr = ffi.cast("struct wlr_drag *", ptr)
+
+        self.focus_event = Signal(ptr=ffi.addressof(self._ptr.events.focus))
+        self.motion_event = Signal(
+            ptr=ffi.addressof(self._ptr.events.motion),
+            data_wrapper=DragMotionEvent,
+        )
+        self.drop_event = Signal(
+            ptr=ffi.addressof(self._ptr.events.drop),
+            data_wrapper=DragDropEvent,
+        )
+        self.destroy_event = Signal(ptr=ffi.addressof(self._ptr.events.destroy))
+
+    @property
+    def icon(self) -> "DragIcon":
+        return DragIcon(self._ptr.icon)
```

Comment:
Do we need to add weakrefs for these to keep the `self._ptr` alive? If so, maybe just add a todo on that for now here and some of the other methods added that use members of the `_ptr`?